src/MongoDB.Driver.Core/Core/Authentication/GssapiAuthenticator.cs

```diff
@@ -177,6 +177,7 @@ private class GssapiMechanism : ISaslMechanism
         {
             // fields
             private readonly bool _canonicalizeHostName;
+            private readonly IDnsResolver _dnsResolver;
```

Comment:
As I see, the dnslookup doc recommends using `lookupClient` as singletone (http://dnsclient.michaco.net/):
    
    It is important to note, that it is highly recommended to instantiate the lookup client only once per application and to always use the same reference (singleton).
    Otherwise, features like result cache and connection pooling will not have any effect which can decrease the application's performance. 

Should we worry about it? Or we can say that `GssapiMechanism` won't be created more than once for Connection and we ok to leave it as is.

Reply:
Not sure, but is the `GssapiAuthenticator` constructed once per socket rather than once per `MongoClient`?

Reply:
>constructed once per socket

currently, yes